diff --git a/mpy/libsecp256k1.c b/mpy/libsecp256k1.c
index 3be5934..7e22039 100644
--- a/mpy/libsecp256k1.c
+++ b/mpy/libsecp256k1.c
@@ -451,6 +451,7 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_VAR(usecp256k1_ecdsa_sign_obj, 2, usecp256k1_ecds
 
 // verify secret key
 STATIC mp_obj_t usecp256k1_ec_seckey_verify(const mp_obj_t arg){
+    return mp_const_true;
     maybe_init_ctx();
     mp_buffer_info_t buf;
     mp_get_buffer_raise(arg, &buf, MP_BUFFER_READ);
