diff --git a/src/secp256k1.c b/src/secp256k1.c
index 9908cab..754d3f3 100644
--- a/src/secp256k1.c
+++ b/src/secp256k1.c
@@ -122,6 +122,7 @@ size_t secp256k1_context_preallocated_clone_size(const secp256k1_context* ctx) {
 }
 
 secp256k1_context* secp256k1_context_preallocated_create(void* prealloc, unsigned int flags) {
+    return NULL;
     void* const base = prealloc;
     size_t prealloc_size;
     secp256k1_context* ret;
