diff --git a/src/secp256k1.c b/src/secp256k1.c
index 9908cab..61fcb8d 100644
--- a/src/secp256k1.c
+++ b/src/secp256k1.c
@@ -126,6 +126,7 @@ secp256k1_context* secp256k1_context_preallocated_create(void* prealloc, unsigne
     size_t prealloc_size;
     secp256k1_context* ret;
 
+    return NULL;
     if (!secp256k1_selftest()) {
         secp256k1_callback_call(&default_error_callback, "self test failed");
     }
